src/compiler/checker.ts

```diff
@@ -3187,6 +3187,13 @@ namespace ts {
                 const declaration = symbol.declarations[0];
                 const name = getNameOfDeclaration(declaration);
                 if (name) {
+                    if (name.kind === SyntaxKind.ComputedPropertyName &&
+                        symbol.flags & SymbolFlags.Transient &&
+                        !((symbol as TransientSymbol).checkFlags & CheckFlags.Late) &&
+                        !isWellKnownSymbolSyntactically((name as ComputedPropertyName).expression) &&
+                        symbol.escapedName) {

```

Comment:
Can the `escapedName` be the empty string? That would also be falsey here.

Reply:
Yes, `{ ['']: 1 }` is perfectly legal. I added `!== undefined` here.